apps/studio/components/interfaces/Auth/Hooks/CreateHookSheet.tsx

```diff
@@ -0,0 +1,425 @@
+                          disabled={field.disabled}
+                        />
+                      </FormControl_Shadcn_>
+                    </FormItemLayout>
+                  )}
+                />
+                <FormField_Shadcn_
+                  key="postgresValues.functionName"
+                  control={form.control}
+                  name="postgresValues.functionName"
+                  render={({ field }) => (
+                    <FormItemLayout
+                      label="Function name"
+                      description="Postgres function which will be called by Supabase Auth each time the hook is triggered."
+                    >
+                      <FormControl_Shadcn_>
+                        <FunctionSelector
+                          size="small"
+                          schema={values.postgresValues.schema}
+                          value={field.value}
+                          onChange={field.onChange}
+                          disabled={field.disabled}
+                        />
+                      </FormControl_Shadcn_>
+                    </FormItemLayout>
+                  )}
+                />
+              </div>
+            ) : (
+              <div className="flex flex-col px-8 gap-4">
+                <FormField_Shadcn_
+                  key="httpsValues.url"
+                  control={form.control}
+                  name="httpsValues.url"
+                  render={({ field }) => (
+                    <FormItemLayout
+                      label="URL"
+                      description="Supabase Auth will send a HTTPS POST request to this URL each time the hook is triggered."
+                    >
+                      <FormControl_Shadcn_>
+                        <Input_Shadcn_ {...field} />
+                      </FormControl_Shadcn_>
+                    </FormItemLayout>
+                  )}
+                />
+                <FormField_Shadcn_
+                  key="httpsValues.secret"
+                  control={form.control}
+                  name="httpsValues.secret"
+                  render={({ field }) => (
+                    <FormItemLayout
+                      label="Secret"
+                      description="It should be a base64 encoded hook secret with a prefix 'v1,whsec_'. 'v1' denotes the signature version, and 'whsec_' signifies a symmetric secret."
+                    >
+                      <FormControl_Shadcn_>
+                        <div className="flex flex-row">
+                          <Input_Shadcn_ {...field} className="rounded-r-none border-r-0" />
+                          <Button
+                            type="default"
+                            size="small"
+                            className="rounded-l-none h-[38px]"
+                            onClick={() => {
+                              const authHookSecret = generateAuthHookSecret()
+                              field.onChange(authHookSecret)
```

Comment:
not sure it makes any difference but you could use 

`form.setValue('httpsValues.secret', authHookSecret)` 

just so you don't hijack the field.onChange as you'd expect that to come from the input, not this button.

Reply:
Fixed.